packages/docusaurus-theme-classic/src/theme/hooks/usePlatformValue.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import useWindowSize from '@theme/hooks/useWindowSize';
+import type {PlatformDependant} from '@docusaurus/utils';
+
+function usePlatformValue<Type>(
+  option: PlatformDependant<Type> | undefined,
+): Type | undefined {
+  const windowSize = useWindowSize();
+  if (option === undefined) {
+    return undefined;
+  }
+  if ((option as Record<string, Type>)[`${windowSize}`] !== undefined) {
```

Comment:
1. Use a type guard to handle the `Type` case before `Record<string, Type>`. and put the type guard in `theme-common/platformDependent`
2. No need to use template strings; `useWindowSize` is already returning a string
3. Need to handle `ssr` as a window size here

Reply:
how should I handle ssr?